### PR TITLE
Add cache for system summary data

### PR DESCRIPTION
**Pull request by Pengpengwanga**
Signed-off-by: Pengpeng Wang <pengpengw@vmware.com>

### CODE REVIEW

File: flowgate-api/src/main/java/com/vmware/flowgate/controller/SystemSummaryController.java [hunk 1]
```diff
@@ -20,7 +24,13 @@
    SummaryService summaryService;
 
    @RequestMapping(value = "/systemsummary", method = RequestMethod.GET)
-   public SystemSummary getAllDashBoardData() {
-      return summaryService.getSystemResult();
+   public SystemSummary getAllDashBoardData( @RequestParam("usecache") boolean usecache) {
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/controller/SystemSummaryController.java]: let's check

File: flowgate-api/src/main/java/com/vmware/flowgate/service/SummaryService.java [hunk 2]
```diff
@@ -35,7 +38,29 @@
...
+
+   public static final String SUMMARY_DATA = "summarydata";
+
+   public SystemSummary getSystemResult(boolean useCache) throws IOException {
+      SystemSummary data = new SystemSummary();
+      ObjectMapper mapper = new ObjectMapper();
+      if(useCache) {
+         if(!redisTemplate.hasKey(SUMMARY_DATA)) {
+            data = generateSummaryData();
+            redisTemplate.opsForValue().set(SUMMARY_DATA, mapper.writeValueAsString(data));
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/SummaryService.java]: you can return here.

File: flowgate-api/src/main/java/com/vmware/flowgate/controller/SystemSummaryController.java (hunk 1 above)
Comment (Pengpengwanga) [reply flowgate-api/src/main/java/com/vmware/flowgate/controller/SystemSummaryController.java]: Done

File: flowgate-api/src/main/java/com/vmware/flowgate/service/SummaryService.java (hunk 2 above)
Comment (Pengpengwanga) [reply flowgate-api/src/main/java/com/vmware/flowgate/service/SummaryService.java]: Done

**Pull request merged**